cmd/operator-sdk/cli/cli.go

```diff
@@ -33,9 +34,9 @@ import (
 )
 
 var commands = []*cobra.Command{
-	// Once the KB CLI is made the default, add the "new" command as a way to
-	// scaffold the legacy project layout and mark "new" as deprecated
-	// new.NewCmd()
+	// The "new" cmd provides a way to scaffold Helm/Ansible projects
+	// from the new CLI.
+	new.NewCmd(),
```

Comment:
The new is showing when we are in the dir of a new layout project. See:

```
Available Commands:
  alpha       Run an alpha subcommand
  build       Compiles code and builds artifacts
  bundle      Manage operator bundle metadata
  completion  Generators for shell completions
  create      Scaffold a Kubernetes API or webhook
  generate    Invokes a specific generator
  help        Help about any command
  init        Initialize a new project
  new         Creates a new operator application
  olm         Manage the Operator Lifecycle Manager installation in your cluster
  version     Prints the version of operator-sdk
``` 

When the new command will be useful for a Golang project with kb layout?

Reply:
In general, it isn't that nice to have a dynamic CLI that is dependent on things like which directory you're in because it makes it harder for users to discover what is possible and to do other things like generate docs.

So I think it is reasonable for the `new` subcommand to be omnipresent in the new CLI until we fully migrate helm and ansible operators over to using `init` and `create api`.

My two cents.

Reply:
OK. Are we not for all the other cases/commands we are just showing that ones which are aligned with the tool?
So, IMO we should keep the consistency. However, it is not a blocker at all and can be checked in a follow up as well.

Reply:
The new default needs to have `new --type=Ansible/Helm` as default. No matter which directory you're in.

It's the old legacy project that's the exception now where we make the switch.

So I think we can live with this since users have scaffolded a Go project with `init` will be at best curious if they see `operator-sdk new` in the CLI. And they can easily see that it's for scaffolding Helm/Ansible projects and not Go projects.